### PR TITLE
HPCC-14711 Fix potential overflows in binary searches

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review very carefully!

File: rtl/eclrtl/rtlkey.cpp [hunk 1]
```diff
@@ -1860,7 +1860,7 @@ void CStringSet::addRange(const void *loval, const void *hival)
         int rc;
         while ((int)a<b)
         {
-            int i = (a+b+1)/2;
+            int i = a+(b+1-a)/2;
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlkey.cpp]: Should these technically be b-a+1 rather than b+1-a ? i.e. can the b+1 overflow and would we care if it did?

Comment (ghalliday) [inline rtl/eclrtl/rtlkey.cpp]: Maybe.  The rules on overflow for unsigned types are well defined, so I they will be equivalent.

Comment (richardkchapman) [inline rtl/eclrtl/rtlkey.cpp]: But b here is an int...

**Pull request merged**